src/kbucket/bucket.rs

```diff
@@ -566,6 +568,15 @@ where
                         };
                     }
                 }
+
+                // If we inserted the node, make sure there is no pending node of the same key. This can
+                // happen when a pending node is inserted, a node gets removed from the bucket, freeing up
+                // space and then re-inserted here.
+                if let Some(pending) = self.pending.as_ref() {
+                    if pending.node.key == node.key {
+                        self.pending = None
+                    }
+                }
```

Comment:
This code is the same as in the disconnected case, so if I understand correctly the logic is: if inserted the pending, remove it? If so what would you say of factoring this out as 
`let insert_result =  match node.status.state { .. };
and later checking if it is Inserted. I might be clearer.

Reply:
we should update the docs to say that if the inserted node matches the pending one, this one is dropped.